models/yang/extensions/openconfig-interfaces-deviation.yang

```diff
@@ -0,0 +1,250 @@
+    deviate not-supported;
+  }
+ 
+  deviation /oc-intf:interfaces/oc-intf:interface/oc-intf:subinterfaces/oc-intf:subinterface/oc-ip:ipv4/oc-ip:addresses/oc-ip:address/oc-ip:vrrp {
+    deviate not-supported;
+  }
+ 
+  deviation /oc-intf:interfaces/oc-intf:interface/oc-intf:subinterfaces/oc-intf:subinterface/oc-ip:ipv4/oc-ip:proxy-arp {
+    deviate not-supported;
+  }
+ 
+  deviation /oc-intf:interfaces/oc-intf:interface/oc-intf:subinterfaces/oc-intf:subinterface/oc-ip:ipv4/oc-ip:neighbors {
+    deviate not-supported;
+  }
+ 
+  deviation /oc-intf:interfaces/oc-intf:interface/oc-intf:subinterfaces/oc-intf:subinterface/oc-ip:ipv4/oc-ip:unnumbered {
+    deviate not-supported;
+  }
+ 
+  deviation /oc-intf:interfaces/oc-intf:interface/oc-intf:subinterfaces/oc-intf:subinterface/oc-ip:ipv4/oc-ip:config {
+    deviate not-supported;
+  }
+ 
+  deviation /oc-intf:interfaces/oc-intf:interface/oc-intf:subinterfaces/oc-intf:subinterface/oc-ip:ipv4/oc-ip:state {
+    deviate not-supported;
+  }
+ 
+  deviation /oc-intf:interfaces/oc-intf:interface/oc-intf:subinterfaces/oc-intf:subinterface/oc-ip:ipv6/oc-ip:addresses/oc-ip:address/oc-ip:state/oc-ip:origin {
+    deviate not-supported;
+  }
+ 
+  deviation /oc-intf:interfaces/oc-intf:interface/oc-intf:subinterfaces/oc-intf:subinterface/oc-ip:ipv6/oc-ip:addresses/oc-ip:address/oc-ip:state/oc-ip:status {
+    deviate not-supported;
+  }
+ 
+  deviation /oc-intf:interfaces/oc-intf:interface/oc-intf:subinterfaces/oc-intf:subinterface/oc-ip:ipv6/oc-ip:addresses/oc-ip:address/oc-ip:vrrp {
+    deviate not-supported;
+  }
+ 
+  deviation /oc-intf:interfaces/oc-intf:interface/oc-intf:subinterfaces/oc-intf:subinterface/oc-ip:ipv6/oc-ip:router-advertisement {
+    deviate not-supported;
+  }
+ 
+  deviation /oc-intf:interfaces/oc-intf:interface/oc-intf:subinterfaces/oc-intf:subinterface/oc-ip:ipv6/oc-ip:neighbors {
+    deviate not-supported;
+  }
+ 
+  deviation /oc-intf:interfaces/oc-intf:interface/oc-intf:subinterfaces/oc-intf:subinterface/oc-ip:ipv6/oc-ip:unnumbered {
+    deviate not-supported;
+  }
+
+  deviation /oc-intf:interfaces/oc-intf:interface/oc-intf:subinterfaces/oc-intf:subinterface/oc-ip:ipv6/oc-ip:config/oc-ip:enabled {
+    deviate replace {
+      default false;
+    }
+  }
+ 
+  deviation /oc-intf:interfaces/oc-intf:interface/oc-intf:subinterfaces/oc-intf:subinterface/oc-ip:ipv6/oc-ip:config/oc-ip:mtu {
+    deviate not-supported;
+  }
+ 
+  deviation /oc-intf:interfaces/oc-intf:interface/oc-intf:subinterfaces/oc-intf:subinterface/oc-ip:ipv6/oc-ip:config/oc-ip:dup-addr-detect-transmits {
+    deviate not-supported;
+  }
+ 
+  deviation /oc-intf:interfaces/oc-intf:interface/oc-intf:subinterfaces/oc-intf:subinterface/oc-ip:ipv6/oc-ip:config/oc-ip:dhcp-client {
+    deviate not-supported;
+  }
+ 
+  deviation /oc-intf:interfaces/oc-intf:interface/oc-intf:subinterfaces/oc-intf:subinterface/oc-ip:ipv6/oc-ip:state {
```

Comment:
Openconfig mandates that every attribute under "config" container should be mirrored in "state" container too.

Here, interfaces/interface/subinterfaces/subinterface/ipv4/config/enabled is present(see line 164),
but  interfaces/interface/subinterfaces/subinterface/ipv4/state is pruned.

This is inconsistent with OpenConfig guidelines.

Reply:
This needs to be corrected

Reply:
Thanks for reviewing this @sayedsaquibbrcm.

The two ipv4 containers are not supported and are marked correctly in the PR. (Attached lines)
1. interfaces/interface/subinterfaces/subinterface/ipv4/config 
https://github.com/sonic-net/sonic-mgmt-common/blob/471426b6e2791652255b2566e8cff31ebd176af9/models/yang/extensions/openconfig-interfaces-deviation.yang#L132 

2. interfaces/interface/subinterfaces/subinterface/ipv4/state 
https://github.com/sonic-net/sonic-mgmt-common/blob/471426b6e2791652255b2566e8cff31ebd176af9/models/yang/extensions/openconfig-interfaces-deviation.yang#L136

The ipv6 config enabled attribute is supported and the corresponding ipv6 state attribute is now marked as supported.
1. interfaces/interface/subinterfaces/subinterface/ipv6/config/enabled   ===> Supported already
2. interfaces/interface/subinterfaces/subinterface/ipv6/state/enabled   ===> Addressed in a new commit and marked as supported.